Junimatic/IndoorPotMachine.cs

```diff
@@ -0,0 +1,195 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.PortableExecutable;
+using System.Xml.Linq;
+using Microsoft.Xna.Framework;
+using StardewValley;
+using StardewValley.Inventories;
+using StardewValley.Objects;
+using StardewValley.TerrainFeatures;
+
+using static NermNermNerm.Stardew.LocalizeFromSource.SdvLocalize;
+
+namespace NermNermNerm.Junimatic
+{
+    internal class IndoorPotMachine : ObjectMachine
+    {
+        private static bool isHarmonyPatchApplied = false;
+
+        internal IndoorPotMachine(IndoorPot machine, Point accessPoint)
+            : base(machine, accessPoint)
+        {
+        }
+
+        public new IndoorPot Machine => (IndoorPot)base.Machine;
+
+        public override bool IsCompatibleWithJunimo(JunimoType projectType) => projectType == JunimoType.Crops;
+
+        public override bool FillMachineFromChest(GameStorage storage, Func<Item, bool> isShinyTest)
```

Comment:
I haven't looked into how shiny test is implemented, but as a future enhancement, could it be piggybacked to implement a no harvest list? A common example is players want to leave flowers in bloom for bees.

Reply:
Hm.  The Shiny list only applies to taking objects from chests to machines.  If it also applied to leaving pots alone, you'd have hell with flowers because of all the different colors (which might conflict.  Not sure...).  But for that specific case, I'd think the solution would be to just make it so the flower pots aren't adjacent to a pathway.

Reply:
If the junimos also handle watering in the future, then it would be advantageous to have them adjacent to a pathway. The different colored flowers all have the same id, they only differ based on the tint field specific to ColoredObject. 

Regardless, after thinking it through a second time, the Shiny list is a fundamentally different concept. The shiny lists bans items from being **placed in** to machines, whereas my idea entails banning items from being **removed from** machines. Even if some functionality could be reused, it's better coding practice to implement a removal ban list as its own thing. (Somewhat unrelated: Another use case for a removal ban list is with crab pots. The farmer needs to remove the "fish" from the crab pot themselves for it to count towards their fishing collection, from my experience. Interestingly, removing the stack of "fish" from the chest the first time does count toward crab pot quests, strangely enough. Same with chicken eggs and gus' collect two dozen eggs special order quest.)

For flowers specifically, I've seen other mods include configuration that let the player ban flowers as a whole from being harvested, instead of individual by item id. Not something that needs to be done now, however. It's really only for bees and honey that this would be useful...and bees only produce outside...and there are other mods that extend junimo hut functionality to harvest outdoor garden pots...